### PR TITLE
Add pod5 datatype

### PR DESCRIPTION
**Pull request by TomHarrop**
I'd like to add the Oxford Nanopore binary [Pod5](https://pod5-file-format.readthedocs.io) datatype to support the dorado wrapper I'm developing [over here](https://github.com/usegalaxy-au/tools-au/pull/112).

That PR also includes tool that can convert `fast5.tar` (already supported by Galaxy) to `pod5`.

pod5 has a [fixed signature](https://pod5-file-format.readthedocs.io/en/latest/SPECIFICATION.html#signature) so the sniffer just checks for that.

Thanks!

## How to test the changes?
(Select all options that apply)
- [X] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -3759,6 +3759,35 @@ def sniff(self, filename: str) -> bool:
         return Fast5Archive.sniff(self, filename)
 
 
+class Pod5(Binary):
+    """
+    Class describing a Pod5 file
```
Comment (bernt-matthias) [inline lib/galaxy/datatypes/binary.py]: Can you add a link to the specification here or in the data types config file?

File: lib/galaxy/datatypes/binary.py [hunk 2]
```diff
@@ -3759,6 +3759,35 @@ def sniff(self, filename: str) -> bool:
...
+
+    >>> from galaxy.datatypes.sniff import get_test_fname
```
Comment (bernt-matthias) [inline lib/galaxy/datatypes/binary.py:3767]: We often also have a negative test. Not sure if needed here.

File: lib/galaxy/datatypes/test/test.pod5
Comment (mvdbeek) [inline lib/galaxy/datatypes/test/test.pod5:1]: Can we make this file smaller ?

Comment (TomHarrop) [reply lib/galaxy/datatypes/test/test.pod5:1]: I subset it to one read and it's still 19K.

File: lib/galaxy/datatypes/binary.py (hunk 2 above)
Comment (TomHarrop) [reply lib/galaxy/datatypes/binary.py:3767]: Done in 385e7ab1e1f35c596b79efa7ddfffe0d4da753e4

File: lib/galaxy/datatypes/binary.py (hunk 1 above)
Comment (TomHarrop) [reply lib/galaxy/datatypes/binary.py]: Done in 385e7ab1e1f35c596b79efa7ddfffe0d4da753e4

**Comment by TomHarrop** [comment]
@bernt-matthias @mvdbeek thanks for the reviews, I've made those changes.

**Comment by TomHarrop** [comment]
I can’t tell for sure if the test failures are related to my changes. Please let me know if I need to fix something in this PR.

**Comment by nsoranzo** [comment]
> I can’t tell for sure if the test failures are related to my changes. Please let me know if I need to fix something in this PR.

All test failures are unrelated to your PR, thanks!

File: lib/galaxy/datatypes/binary.py [hunk 3]
```diff
@@ -3759,6 +3759,34 @@ def sniff(self, filename: str) -> bool:
...
+    def sniff(self, filename: str) -> bool:
+        expected_signature = bytes([0x8B, 0x50, 0x4F, 0x44, 0x0D, 0x0A, 0x1A, 0x0A])
+        try:
+            with open(filename, "rb") as f:
+                first_8_bytes = f.read(8)
+                f.seek(-8, 2)
+                last_8_bytes = f.read(8)
+                return first_8_bytes == expected_signature and last_8_bytes == expected_signature
+        except Exception:
+            return False
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/binary.py]: You can simplify the code by removing the try/except as by default Galaxy consider as "false" an unhandled exception in a sniffer.

Comment (TomHarrop) [reply lib/galaxy/datatypes/binary.py]: Thanks, I've removed that.

**Review by nsoranzo — APPROVED** [review]

**Comment by bgruening** [comment]
@TomHarrop you can backport datatypes to the stable release, eg. 24.1

**Comment by TomHarrop** [comment]
Thanks @bgruening I tried to cherry-pick the commits into the release_24.1 branch [here](https://github.com/galaxyproject/galaxy/pull/18507).

**Pull request merged**